utils/cpu.go

```diff
@@ -92,9 +113,103 @@ func (c *CPUStats) monitorCPULoad() {
 				c.warningThrottle(func() { logger.Infow("high cpu load", "load", 1-idleRatio) })
 			}
 
-			if c.updateCallback != nil {
-				c.updateCallback(idle)
+			if c.idleCallback != nil {
+				c.idleCallback(idle)
+			}
+		}
+	}
+}
+
+func (c *CPUStats) monitorProcCPULoad() {
+	numCPU := c.platform.numCPU()
+	fs, err := procfs.NewFS(procfs.DefaultMountPoint)
+	if err != nil {
+		logger.Errorw("failed read proc fs", err)
+		return
+	}
+
+	self, err := fs.Self()
+	if err != nil {
+		logger.Errorw("failed to read self", err)
+		return
+	}
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	var prevTotalTime float64
+	var prevStats map[int]procfs.ProcStat
+	for {
+		select {
+		case <-c.closeChan:
+			return
+		case <-ticker.C:
+			nextStats := make(map[int]procfs.ProcStat)
+			procs, err := procfs.AllProcs()
+			if err != nil {
+				logger.Errorw("failed to read processes", err)
+				continue
+			}
+
+			total, err := fs.Stat()
```

Comment:
This seems to get data from /proc/stat. Do you know if that returns data for the pod only or for the whole system? I believe it does matter here as we are normalizing the result to `platform.numCPU()` which will return the amount of cores associated to the pod.

Reply:
Ah, you're right. Looks like this will cause issues
https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giad069/7250999